src/memory/__init__.py

```
@@ -28,15 +31,119 @@ async def setup(self) -> None:  # pragma: no cover - no-op
+            )
+            # from_conn_string 가 반환하는 contextmanager 안에서 실제 saver 를 획득
+            with PostgresSaver.from_conn_string(settings.POSTGRES_URI) as saver:  # type: ignore[attr-defined]
+                # 초기 실행 시 필요한 checkpoints 테이블 등이 없다면 생성한다.
+                try:
+                    saver.setup()  # type: ignore[attr-defined]
+                    logger.info("PostgreSQL checkpointer schema setup completed")
+                except Exception as setup_error:  # pragma: no cover - fallback path
+                    logger.error(
+                        "PostgreSQL checkpointer setup failed, falling back to MemorySaver: %s",
+                        setup_error,
+                    )
```

Ruff (0.14.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)